src/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+
+add_library(pico_ice_sdk INTERFACE)
```

Comment:
I did not use interfaces before, but it looks like Raspberry Pi really likes them.
It might be what permits to split the main CMakeLists.txt in multiple parts like you are offering here.